packages/jsii-pacmak/lib/targets/go/types/class.ts

```diff
@@ -12,15 +10,21 @@ import { EmitContext } from '../emit-context';
  * GoClass wraps a Typescript class as a Go custom struct type
  */
 export class GoClass extends GoStruct {
-  public readonly methods: ClassMethod[];
+  public readonly methods: ClassMethod[] = [];
+  public readonly staticMethods: StaticMethod[] = [];
```

Comment:
Since `StaticMethod` extends `ClassMethod`, I wonder if the distinction here is necessary... Basically `StaticMehtod`s are `ClassMethod`s, and the code generation is the same, etc...

The only apparent benefit from separating them is you can guarantee `static` methods are generated strictly before all `instance` methods... but is this worth it?

I mean - I don't mind it being this way at all (especially as it's a pretty damned effective way to have `static`s all grouped in one block while avoiding to resort to `sort`), just kinda curious if there is another reason?

Reply:
Mostly bc the emit logic is different, so i thought it'd be best to have the child class override it.